Unmarshal Error for GetChatMenuButton

Hello! I got an error like below when I call Do() of GetChatMenuButton():

```
panic: unmarshal: json: cannot unmarshal object into Go value of type tg.MenuButton [recovered]
```

--- a/examples/webapps/main.go
+++ b/examples/webapps/main.go
@@ -68,6 +68,22 @@ func run(ctx context.Context) error {
 	}
 	log.Printf("auth as https://t.me/%s", me.Username)
 
+	if err := client.SetChatMenuButton().MenuButton(&tg.MenuButtonWebApp{
+		Text: "Open Web App",
+		WebApp: tg.WebAppInfo{
+			URL: flagBaseURL + "/webapp",
+		},
+	}).DoVoid(ctx); err != nil {
+		return fmt.Errorf("set menu button: %w", err)
+	}
+
+	menuButton, err := client.GetChatMenuButton().Do(ctx)
+	if err != nil {
+		return fmt.Errorf("get menu button: %w", err)
+	}
+
+	log.Printf("webapp menu button: %#v", menuButton.WebApp)
+
 	router := newRouter(flagBaseURL)
 
 	webhook := tgb.NewWebhook(router, client, flagBaseURL+"/webhook",

--- a/methods_gen.go
+++ b/methods_gen.go
@@ -4159,13 +4159,13 @@ func (call *SetChatMenuButtonCall) MenuButton(menuButton MenuButton) *SetChatMen
 // Use this method to get the current value of the bot's menu button in a private chat, or the default menu button
 // Returns MenuButton on success.
 type GetChatMenuButtonCall struct {
-	Call[MenuButton]
+	Call[MenuButtonOneOf]
 }
 
 // NewGetChatMenuButtonCall constructs a new GetChatMenuButtonCall with required parameters.
 func NewGetChatMenuButtonCall() *GetChatMenuButtonCall {
 	return &GetChatMenuButtonCall{
-		Call[MenuButton]{
+		Call[MenuButtonOneOf]{
 			request: NewRequest("getChatMenuButton"),
 		},
 	}

--- a/types_gen.go
+++ b/types_gen.go
@@ -1878,7 +1878,7 @@ type InlineQueryResultsButton struct {
 	// Label text on the button
 	Text string `json:"text"`
 
-	// Optional. Description of the Web App that will be launched when the user presses the button. The Web App will be able to switch back to the inline mode using the method web_app_switch_inline_query inside the Web App.
+	// Optional. Description of the Web App that will be launched when the user presses the button. The Web App will be able to switch back to the inline mode using the method switchInlineQuery inside the Web App.
 	WebApp *WebAppInfo `json:"web_app,omitempty"`
 
 	// Optional. Deep-linking parameter for the /start message sent to the bot when a user presses the button. 1-64 characters, only A-Z, a-z, 0-9, _ and - are allowed.Example: An inline bot that sends YouTube videos can ask the user to connect the bot to their YouTube account to adapt search results accordingly. To do this, it displays a 'Connect your YouTube account' button above the results, or even before showing any. The user presses the button, switches to a private chat with the bot and, in doing so, passes a start parameter that instructs the bot to return an OAuth link. Once done, the bot can offer a switch_inline button so that the user can easily return to the chat where they wanted to use the bot's inline capabilities.

--- a/types_gen_ext.go
+++ b/types_gen_ext.go
@@ -868,6 +868,38 @@ func (button MenuButtonWebApp) MarshalJSON() ([]byte, error) {
 	return json.Marshal(alias(button))
 }
 
+// MenuButtonOneOf contains one of MenuButton implementations.
+// It's used for proper JSON marshaling.
+type MenuButtonOneOf struct {
+	Default  *MenuButtonDefault
+	Commands *MenuButtonCommands
+	WebApp   *MenuButtonWebApp
+}
+
+func (button *MenuButtonOneOf) UnmarshalJSON(v []byte) error {
+	var partial struct {
+		Type string `json:"type"`
+	}
+
+	if err := json.Unmarshal(v, &partial); err != nil {
+		return fmt.Errorf("unmarshal MenuButtonOneOf partial: %w", err)
+	}
+
+	switch partial.Type {
+	case "default":
+		button.Default = &MenuButtonDefault{}
+		return json.Unmarshal(v, button.Default)
+	case "commands":
+		button.Commands = &MenuButtonCommands{}
+		return json.Unmarshal(v, button.Commands)
+	case "web_app":
+		button.WebApp = &MenuButtonWebApp{}
+		return json.Unmarshal(v, button.WebApp)
+	default:
+		return fmt.Errorf("unknown MenuButtonOneOf type: %s", partial.Type)
+	}
+}
+
 // MessageType it's type for describe content of Message.
 type MessageType int
 

--- a/types_gen_ext_test.go
+++ b/types_gen_ext_test.go
@@ -6,6 +6,7 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestPeerIDImpl(t *testing.T) {
@@ -1099,4 +1100,37 @@ func TestStickerType_UnmarshalText(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, test.Want, e)
 	}
+}
+
+func TestMenuButtonOneOf_UnmarshalJSON(t *testing.T) {
+	t.Run("Commands", func(t *testing.T) {
+		var b MenuButtonOneOf
+
+		err := b.UnmarshalJSON([]byte(`{"type": "commands"}`))
+		require.NoError(t, err)
+
+		require.NotNil(t, b.Commands)
+		assert.Equal(t, "commands", b.Commands.Type)
+	})
+
+	t.Run("Default", func(t *testing.T) {
+		var b MenuButtonOneOf
+
+		err := b.UnmarshalJSON([]byte(`{"type": "default"}`))
+		require.NoError(t, err)
+
+		require.NotNil(t, b.Default)
+		assert.Equal(t, "default", b.Default.Type)
+	})
+
+	t.Run("WebApp", func(t *testing.T) {
+		var b MenuButtonOneOf
+
+		err := b.UnmarshalJSON([]byte(`{"type": "web_app", "text": "12345"}`))
+		require.NoError(t, err)
+
+		require.NotNil(t, b.WebApp)
+		assert.Equal(t, "web_app", b.WebApp.Type)
+		assert.Equal(t, "12345", b.WebApp.Text)
+	})
 }

